medicines/doc-index-updater/src/create_manager/sanitiser.rs

```diff
@@ -0,0 +1,114 @@
+#[derive(Clone)]
+pub struct SanitisedString {
+    inner: String,
+}
+
+impl From<String> for SanitisedString {
+    fn from(inner: String) -> Self {
+        Self { inner }
+    }
+}
+
+impl From<&String> for SanitisedString {
+    fn from(inner: &String) -> Self {
+        Self::from(inner.to_owned())
+    }
+}
+
+impl From<&str> for SanitisedString {
+    fn from(inner: &str) -> Self {
+        Self::from(inner.to_owned())
+    }
+}
+
+impl ToString for SanitisedString {
+    fn to_string(&self) -> String {
+        self.inner
+            .replace(|c: char| !c.is_ascii(), "")
+            .replace("\n", " ")
+            .trim()
+            .to_string()
+    }
+}
+
+#[derive(Clone)]
+pub struct VecSanitisedString {
+    inner: Vec<String>,
+}
+
+impl From<Vec<String>> for VecSanitisedString {
+    fn from(inner: Vec<String>) -> Self {
+        Self { inner }
+    }
+}
+
+// impl Iterator for VecSanitisedString {
+//     type Item = SanitisedString;
+//     fn next(&mut self) -> Option<Self::Item> {
+//         if let Some(a) = self.iter {
+//             //
+//         } else {
+//             self.iter = self.inner.iter();
+//         }
+
+//         if let Some(a) = self.iter.next() {
+//             Some(SanitisedString::from(a))
+//         } else {
+//             self.iter = None;
+//             None
+//         }
+//     }
+// }
+
+impl VecSanitisedString {
+    pub fn to_vec_string(&self) -> Vec<String> {
+        self.clone()
+            .iter()
+            .map(|s: &String| SanitisedString::from(s).to_string())
+            .collect::<Vec<String>>()
+    }
+
+    pub fn join(&self, sep: &str) -> String {
+        self.to_vec_string().join(sep)
+    }
+
+    pub fn to_json(&self) -> String {
+        serde_json::to_string(&self.to_vec_string()).expect("Couldn't create JSON array.")
+    }
+
+    pub fn iter(&self) -> std::slice::Iter<'_, String> {
+        self.inner.iter()
+    }
+}
+
+impl Default for VecSanitisedString {
+    fn default() -> Self {
+        Self {
+            inner: Default::default(),
+        }
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use super::*;
+
+    #[test]
+    fn sanitise_remove_newline() {
+        assert_eq!(
+            SanitisedString::from("newline\ntest").to_string(),
+            "newline test"
+        );
+    }
+    #[test]
+    fn sanitise_remove_non_ascii() {
+        assert_eq!(
+            SanitisedString::from("emoji🙂 ∫test").to_string(),
+            "emoji test"
```

Comment:
Would be nice to include an accented character here to double-check that's working. Maybe some common punctuation outside of ASCII which could easily end up being entered into a system: a non-breaking space, en dash, left and right quotation marks, etc.

Reply:
great call, pushed one now